minesweeper.py

```diff
@@ -18,7 +21,7 @@
 
 GameStatus = Enum("GameStatus", "in_progress, game_lost, game_won")
 try:
-    with open(image_dir / "game_data.json", "r") as source:
+    with open(dirs.user_config_dir / "game_data.json", "r") as source:
```

Comment:
- `user_config_dir` is a string, not a `pathlib.Path` object, so this is `str / str`. That's an error. One of the two has to be a `pathlib.Path` object for this to work. 
- The directory does not exist when the program is started for the first time. You need to create the directory if it doesn't exist already.

You could have spotted both problems by actually trying to run the code.